notebooks/106-auto-device/106-auto-device.ipynb

```diff
@@ -0,0 +1,485 @@
+    "output_layer = compiled_model.output(0)\n",
+    "\n",
+    "# get the first inference result\n",
+    "results = compiled_model([input_image])[output_layer]\n",
+    "\n",
+    "\n",
+    "# Get 1st inference, time point 2\n",
+    "auto_fil_end_time = time.perf_counter()\n",
+    "auto_fil_span = auto_fil_end_time - auto_load_start_time\n",
+    "print(f\"Loaded model to AUTO and get first inference in {auto_fil_end_time-auto_load_start_time:.2f} seconds.\")\n",
+    "del compiled_model"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1a238bbe-07f0-4623-addf-e2ef58216b26",
+   "metadata": {},
+   "source": [
+    "### First inference latency benefit "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "d6811f16-29a4-450a-81eb-fe4eb7909f9c",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Output the latency difference\n",
+    "device_list = [\"GPU\", \"AUTO\"]\n",
+    "load_and_fil_list = [gpu_fil_span, auto_fil_span]\n",
+    "plt.barh(range(len(load_and_fil_list)), load_and_fil_list, tick_label=device_list)\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "0a2de05c-6b75-4bc2-814a-08aa4924c2a7",
+   "metadata": {},
+   "source": [
+    "## Performance hint\n",
+    "The next highlight is the differentiation of performance hint with AUTO. By specifying LATENCY hint or THROUGHTPUT hint, AUTO demonstrate significant performance results towards the desired metric. THROUGHTPUT hint delivers much higher frame per second (FPS) performance than LATENCY hint. In contrast, the LATENCY hint delivers much lower latency than THROUGHTPUT hint. Notice that the hints do not require low-level device-specific settings, and are also completely portable between the devices, which allows the AUTO just to expedite the hint value directly to the selected device.\n",
+    "\n",
+    "### Inference with THROUGHTPUT hint\n",
+    "\n",
+    "Loop for the inference and output the FPS/Latency for each @period_seconds sencods"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "94b8e571",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# output period (seconds)\n",
+    "period_seconds = 10\n",
+    "end_after_periods = 6  # Total time @period_seconds x @end_after_periods\n",
+    "\n",
+    "class InferContext:\n",
+    "    def __init__ (self):\n",
+    "        self.reset()\n",
+    "        \n",
+    "    def reset(self):\n",
+    "        self.period_fps = 0\n",
+    "        self.period_latency = 0\n",
+    "        self.period_start_time = time.perf_counter()\n",
+    "        self.period_count = 0\n",
+    "        self.latency_list = []\n",
+    "        self.overall_latency_list = []\n",
+    "        self.feed_inference = True\n",
+    "\n",
+    "class InferJob:\n",
+    "    def __init__ (self, id, context):\n",
+    "        self.id = id\n",
+    "        self.context = context\n",
+    "\n",
```

Comment:
Please include docstring format and type hints, for all classes, functions, or methods.

![image](https://user-images.githubusercontent.com/10940214/161100871-e61bf5f3-ce6e-4b1b-a615-abf2dcc0ec59.png)